New pattern submission: Amazon S3, SQS, Lambda, and Bedrock CDK Python (#2248)
aws-samples/serverless-patterns: s3-sqs-lambda-bedrock-s3-cdk-python/README.md

> # Conversation Summarization: Leveraging Amazon S3, SQS, Lambda, and Bedrock\

Author: change implemented

## Prerequisite
Amazon Bedrock users need to request access to models before they are available for use. If you want to add additional models for text, chat, and image generation, you need to request access to models in Amazon Bedrock. Please refer to the link below for instruction:
[Model access](https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html).

## Deployment Instructions

1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
    ``` 
    git clone https://github.com/aws-samples/serverless-patterns
    ```
2. Change directory to the pattern directory:
    ```
    cd s3-sqs-lambda-bedrock-s3-cdk-python
    ```
3. Create a virtual environment for Python:
   ```
   python3 -m venv .venv
   ```
4. Activate the virtual environment
   ```
   source .venv/bin/activate
   ```
   For a Windows platform, activate the virtualenv like this:
   ```
   .venv\Scripts\activate.bat
5. Install the Python required dependencies:
   ```
   pip install -r requirements.txt
   ```
6. Review the CloudFormation template the cdk generates for you stack using the following AWS CDK CLI command:
   ```

>    cdk synth

Author: cdk.json added to resolve this error

   ```
7. Run the command below to bootstrap your account CDK needs it to deploy
    ```
    cdk bootstrap
    ```
8. From the command line, use CDK to deploy the AWS resources for the pattern. You'll be prompted to approve security related changes during the deployment.
    ```
    cdk deploy
    ```

## How it works

> * Upload a support conversation history input file to the input S3 Bucket, triggering an 'OBJECT_CREATED' event notification to an SQS Queue. Upon receipt, the SQS Queue triggers a Lambda function which will fetch the content of the file using Bucket and Object key info in the event. The Lambda constructs a prompt containing the support conversation and invokes Amazon Titan Text Express from Bedrock to generate conversation summary and sentiment in JSON format. Finally, the Lambda uploads this summary to the summary S3 Bucket.

Reviewer: ```suggestion
* Upload a support conversation history input file to the input S3 Bucket, triggering an 'OBJECT_CREATED' event notification to an SQS Queue. A Lambda function receives the message from the queue and fetches the content  using the Bucket and Object key information in the event. The function constructs a prompt containing the support conversation and invokes Amazon Titan Text Express from Bedrock to generate a conversation summary and sentiment in JSON format. Finally, the Lambda function uploads the output to the summary S3 Bucket.
```

## Testing

> 1. Navigate to the S3 bucket containing key 's3sqslambdabedrocks3cdkpythons-inputbucket**' along with your stack name and a hash key

Reviewer: ```suggestion
1. Navigate to the S3 bucket with a name containing 's3sqslambdabedrocks3cdkpythons-inputbucket**' 
```

2. Upload input file containing conversation history to the bucket (sample input files are available under ./lambda folder)

> 3. Wait for the AWS Lambda function to invoke the Amazon Bedrock model to get the sentiment and summary of the conversation. The Response will be uploaded to 's3sqslambdabedrocks3cdkpytho-summarybucket' using the same name of the input file uploaded earlier.

Reviewer: ```suggestion
3. Wait for Amazon Bedrock to produce the sentiment and summary of the conversation. The Response will be uploaded to 's3sqslambdabedrocks3cdkpytho-summarybucket' using the same name of the input file uploaded earlier.
```
